transcom/mymove
# B-18855 Delivery out of SIT FSC Calculation Issue

[B-18855](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-18855)

## Steps used to test
1. Use the "HHGMoveInSITEndsToday" option from the Test harness list.
2. Copy the move locator from the JSON object displayed (see Fig1 below)
3. Navigate back to the sign in page for milmove office users
4. Sign in as a Multi-role user under the KKFA GBLOC.
5. Start by clicking the Sign in button for a Prime user.
6. Search for the move by the move locator you copied earlier. Then click on that move in the list.
7. Click the "Create payment request" button.
8. From the list of service items you are able to select from you should see an item called "Domestic origin SIT fuel surcharge". Put a check in the checkbox for that item ("Add to payment request").
9. In the input field labeled "Weight Billed (if different from shipment weight)", enter a weight such as 10000 or greater.
Note: Since by default the shipment will have less than 5000 set for each of the estimated and actual weights on this shipment, entering 10000 should produce a FSC Multiplier of "0.0006255". For further details, see the chart called Fig2 below.
10. Click the "Submit payment request" button at the bottom of the form.
11. Click the "Change user role" link at the top of the page.
12. Make the appropriate selection to log in as a TOO user.
13. Search for the same move locator and click on the associated move.
14. Click the "Payment requests" tab.
Note: there is a known issue when certain conditions are met that the TOO must review weights associated. Steps 15-16 are to workaround this known issue.
15. Click the "Review Weights" button.
16. Click the "Payment requests" tab again.
17. Click the "Review Service Items" button.
18. Observe that the service item called "Domestic origin SIT fuel surcharge" has a "Show calculations" link. Click that link to show the "Weight-based distance multiplier". Observe that the correct multiplier is shown in accordance with the following chart in Fig2. The multiplier should take Billed Weight entered when the payment request is created as priority over the actual or estimated weights entered throughout the shipment creation process.

You may also test the DB entered the correct data by using the following DB query. Make sure to replace the locator (QCDHDH) with your move's locator.
```
select psip.value from payment_service_item_params psip
join payment_service_items psi on psip.payment_service_item_id = psi.id
join payment_requests pr on psi.payment_request_id = pr.id
join service_item_param_keys sipk on psip.service_item_param_key_id = sipk.id
join moves m on pr.move_id = m.id
where m."locator" = 'QCDHDH' and sipk.key = 'WeightBilled'
```

## Fig1
![Screenshot 2024-03-25 at 2 13 29 PM](https://github.com/transcom/mymove/assets/147739091/e18e2206-9755-4d71-9a4a-8e99127cd97b)

## Fig2
weight billed will be referred to as "WB" in the following figure.
WB <= 5000: multiplier is "0.000417"
WB between 5001-10000: multiplier is "0.0006255"
WB between 10001-24000: multiplier is "0.000834"
WB 24001+: multiplier is "0.00139"
